util/progress/progressui/colors.go

```diff
@@ -0,0 +1,142 @@
+package progressui
+
+import (
+	"encoding/csv"
+	"strconv"
+	"strings"
+
+	"github.com/morikuni/aec"
+	"github.com/sirupsen/logrus"
+)
+
+var termColorMap = map[string]aec.ANSI{
+	"default": aec.DefaultF,
+
+	"black":   aec.BlackF,
+	"blue":    aec.BlueF,
+	"cyan":    aec.CyanF,
+	"green":   aec.GreenF,
+	"magenta": aec.MagentaF,
+	"orange":  aec.Color8BitF(aec.NewRGB8Bit(255, 140, 0)),
+	"red":     aec.RedF,
+	"white":   aec.WhiteF,
+	"yellow":  aec.YellowF,
+
+	"light-black":   aec.LightBlackF,
+	"light-blue":    aec.LightBlueF,
+	"light-cyan":    aec.LightCyanF,
+	"light-green":   aec.LightGreenF,
+	"light-magenta": aec.LightMagentaF,
+	"light-orange":  aec.Color8BitF(aec.NewRGB8Bit(255, 165, 0)),
+	"light-red":     aec.LightRedF,
+	"light-white":   aec.LightWhiteF,
+	"light-yellow":  aec.LightYellowF,
+}
+
+func setUserDefinedTermColors(colorsEnv string) {
+	fields := readBuildkitColorsEnv(colorsEnv)
+	if fields == nil {
+		return
+	}
+	for _, field := range fields {
+		parts := strings.SplitN(field, "=", 2)
+		if len(parts) != 2 {
+			logrus.Warnf("Could not parse BUILDKIT_COLORS component: %s", strings.Join(parts, "="))
+			continue
+		}
+		if strings.Contains(parts[1], "=") {
+			logrus.Warnf("Could not parse BUILDKIT_COLORS component: %s", strings.Join(parts, "="))
+			continue
+		}
+		k := strings.ToLower(parts[0])
+		v := parts[1]
+		c, ok := termColorMap[strings.ToLower(v)]
+		if ok {
+			parseKeys(k, c)
+		} else {
+			if strings.Contains(v, ",") {
+				rgbFields := readRGB(v)
+				if rgbFields == nil {
+					continue
+				}
+				ok = isValidRGB(rgbFields)
+				if ok {
+					p1, _ := strconv.Atoi(rgbFields[0])
+					p2, _ := strconv.Atoi(rgbFields[1])
+					p3, _ := strconv.Atoi(rgbFields[2])
+					c = aec.Color8BitF(aec.NewRGB8Bit(uint8(p1), uint8(p2), uint8(p3)))
+					parseKeys(k, c)
+				}
```

Comment:
suggestion: I think a lot of this logic could be bundled into `readRGB`:

```go
if c, ok := termColorMap[strings.ToLower(v)]; ok {
    parseKeys(k, c)
} else if c, ok := readRGB(v); ok {
    parseKeys(k, c)
}
```

Reply:
clean up